src/symmetry.jl

```diff
@@ -66,13 +66,32 @@ end
 Filter out the symmetry operations that respect the symmetries of the discrete BZ grid
 """
 function symmetries_preserving_kgrid(symmetries, kcoords)
+    kcoords_normalized = normalize_kpoint_coordinate.(kcoords)
+    T = eltype(kcoords[1])
+    tol = T <: Rational ? 0 : sqrt(eps(T))
+    is_approx_in(x, X, tol) = any(y -> norm(x-y) ≤ tol, X)
     function preserves_grid(S)
-        all(normalize_kpoint_coordinate(S * k) in kcoords
-            for k in normalize_kpoint_coordinate.(kcoords))
+        all(is_approx_in(normalize_kpoint_coordinate(S * k), kcoords_normalized, tol)
+            for k in kcoords_normalized)
     end
     filter(symop -> preserves_grid(symop[1]), symmetries)
 end
 
+
+"""
+Find the subset of symmetries compatible with the grid induced by the given kcoords and ksymops
+"""
+function symmetries_preserving_kcoords_ksymops(symmetries, kcoords, ksymops)
+    all_kcoords = []
```

Comment:
typing

Reply:
do you know of a good way to do this without mutation?

Reply:
Yes. Apply a map to Iterators.product.

Reply:
ok then, just keep it with the mutation.